x-pack/libbeat/management/managerV2.go

```diff
@@ -49,9 +53,11 @@ type BeatV2Manager struct {
 	errCanceller context.CancelFunc
 
 	// track individual units given to us by the V2 API
-	mx      sync.Mutex
-	units   map[unitKey]*client.Unit
-	actions []client.Action
+	mx          sync.Mutex
+	units       map[unitKey]*client.Unit
+	actions     []client.Action
+	forceReload atomic.Bool
+	reloadLock  sync.Mutex
```

Comment:
I'm being very conservative here. I still need to check/understand if this is needed.